src/model.js

```diff
@@ -1,62 +1,99 @@
-const isFunction = o => typeof o === 'function'
+const { assign, keys } = Object
 
-export const Node = funcOrType => body => ({ type: isFunction(funcOrType) ? funcOrType.name : funcOrType, ...body })
+//===============================================================================================================================
+// BEHAVIOUR
+//===============================================================================================================================
+
+const nodeBehaviour = {
+  is(typeOrCategory) {
+    return typeOrCategory.toString().split(',').some(type => this.type === type)
+  },
+
+  copy(diff) {
+    const clone = { ...this }
+    for (const key in diff) {
+      clone[key] = diff[key] instanceof Function ? diff[key](clone[key]) : diff[key]
+    }
+    return clone
+  }
+}
+
+export const traverse = configuration => assign(model => {
+  const applicableTxs = keys(configuration).filter(key => key.split(',').some(type => model.type === type))
+  if (applicableTxs.length < 1) throw new TypeError(`No matching traverse strategy for ${model.type} node`)
+  return configuration[applicableTxs[0]](model)
+}, configuration)
```

Comment:
I don't really have a concrete use case yet, but my idea is that, even if most of the time you will let the traverse function choose what configuration to use, sometimes you may also want to manually choose a path. For example, for compiling sentences you could do:

```javascript
compile = traverse({
  [Sentence]: node => compile[node.type](node) + ';',  // This entry would be called for any sentence
  [Assignment]: ({var, val}) => `${var} = ${val} // This could only be called manually, since Sentence entry hides it
  ...
})
```

Also, this allows us to define methods that not necessarily are bound to a node type or category but are still part of the "reified operation":

```javascript
const doStuff = traverse({
  notANode: x => ....
})

doStuff.notANode(something)
```